### ISSUE TITLE
error prediction with global model  [BUG]

### ISSUE THREAD

**Issue by nyk2001** [sktime/sktime#3803]
Hi,

I am using **hierarchical** data (sales forecast data from Kaggle competition) to train a global model. 

There are 10 stores and each store has 50 items. My train dataset (X_train) looks like as follow:

![image](https://user-images.githubusercontent.com/3079053/202317601-927cfc69-acdf-48c0-956c-70050b0499d3.png)

My train label dataset (prediction label is **sales**) is as follows:

![image](https://user-images.githubusercontent.com/3079053/202317747-e7f472da-cc43-4652-b7a2-db0e5baa444f.png)

I have successfully trained a global model. The code is:

`def get_transformed_target_forecaster(alpha):
    params ={
        'n_estimators': 3000,
        'max_depth': 16,
        'num_leaves': 2**4,
        'learning_rate': 0.01,
        'boosting_type': 'dart',
        'objective':'quantile'
    }

    #Initialize Light GBM Regressor   
    regressor = lgb.LGBMRegressor(alpha=alpha, **params)
    
    #-----------------------Forecaster Pipeline-----------------
  
    pipe = ForecastingPipeline(
        steps=[
            ("deseasonalise", Deseasonalizer(model="additive", sp=12)),
            ("detrend", Detrender(forecaster=PolynomialTrendForecaster(degree=1))),
             ("forecast", make_reduction(estimator=regressor, strategy="recursive", pooling='global'))
            ]
        )

    
    return pipe

forecaster = get_transformed_target_forecaster(0.9)

# Fit transformer
forecaster_model = forecaster.fit(y_train.to_pandas(), X_train.to_pandas())
`
The output of fitted model is 

![image](https://user-images.githubusercontent.com/3079053/202318158-aa95fefa-b6a4-4dcd-9f2c-442f43872d4b.png)

I am trying to make predictions but getting an error message 

![image](https://user-images.githubusercontent.com/3079053/202320774-037cb023-72aa-4461-a4d9-4dbbfc9a5da8.png)

It is a bug or intended behaviour ?  If it is not a bug can someone help me to resolve this issue ? 

Is there a way I can define Forecast Horizon using the following code : 

`fh_abs = ForecastingHorizon(y_test.index, is_relative=False) `

**Comment by fkiraly** [comment]
can you provide us with your code, full imports, and optimally using dummy data from `sktime`?

Please also spell out your package versions, instructions are in the bug report template.

**Comment by nyk2001** [comment]
Hi @fkiraly , 

The versions of libraries are as follows:

![image](https://user-images.githubusercontent.com/3079053/203039820-47159ad7-8339-4725-9e6e-ab9aa94605bd.png)

![image](https://user-images.githubusercontent.com/3079053/203039936-8409e3a1-4133-4f17-b733-8276c1c91913.png)

The dummy data is taken from the following Kaggle competition:
https://www.kaggle.com/competitions/demand-forecasting-kernels-only/data

I am using Azure Databricks to train the model. So, I convert PySpark dataframe to Pandas before calling the SkTime function. 

The code is as follows:
`import lightgbm as lgb
import statsmodels.api as sm
import pandas as pd
from sktime.forecasting.model_selection import temporal_train_test_split
from sktime.transformations.series.detrend import Deseasonalizer, Detrender
from sktime.forecasting.trend import PolynomialTrendForecaster
from sktime.forecasting.compose import TransformedTargetForecaster, make_reduction
from sklearn.pipeline import make_pipeline
from sktime.forecasting.compose import ForecastingPipeline, make_reduction

def get_transformed_target_forecaster(alpha):
    params ={'n_estimators': 3000,
        'max_depth': 16,
        'num_leaves': 2**4,
        'learning_rate': 0.01,
        'boosting_type': 'dart',
        'objective':'quantile'}

    #Initialize Light GBM Regressor   
    regressor = lgb.LGBMRegressor(alpha=alpha, **params)
    
    #-----------------------Forecaster Pipeline-----------------
  
    pipe = ForecastingPipeline(
        steps=[
            ("deseasonalise", Deseasonalizer(model="additive", sp=12)),
            ("detrend", Detrender(forecaster=PolynomialTrendForecaster(degree=1))),
             ("forecast", make_reduction(estimator=regressor, strategy="recursive", pooling='global'))
            ]
        )

    
    return pipe

forecaster = get_transformed_target_forecaster(0.9)

# Fit transformer
forecaster_model = forecaster.fit(y_train.to_pandas(), X_train.to_pandas())`

The model trains but I cannot make predictions. On the other hand, if I just replace forecaster with Prophet then everything works fine. 

Please let me know if you need more information.

**Comment by nyk2001** [comment]
[train.csv](https://github.com/sktime/sktime/files/10056515/train.csv)

@fkiraly attached is the dataset I used
